packages/react-dom-bindings/src/server/ReactDOMFloatServer.js

```diff
@@ -285,7 +285,17 @@ function preinit(href: string, options: PreinitOptions) {
     // simply return and do not warn.
     return;
   }
-  const resources = currentResources;
+  preinitImpl(href, options, currentResources);
+}
+
+// On the server, preinit may be called outside of render when sending an
+// external SSR runtime as part of the initial resources payload. Since this
+// is an internal React call, we do not need to use the resources stack.
+export function preinitImpl(
+  href: string,
+  options: PreinitOptions,
+  resources: Resources,
```

Comment:
nit, stylistically I've been putting resources first in function argument order because it is very much not optional and options sort of is (i mean it isn't in this case but conceptually it could be).